### PR TITLE
Add updateinfo.xml transitional schema

### PR DESCRIPTION
**Pull request by stewartsmith**
The permissive schema validates all sorts of oddities that are near universally restricted to EoL versions of Linux distributions.

A transitional schema aims for clarifying what is valid in a modern context and is a step towards a stricter and fully defined format for updateinfo.

---

A further TODO is to re-jig the split of some of the documentation between the permissive and transitional schemas.

A strict schema is certainly a TODO item, but requires further collaboration between distributions as there are a lot of effectively meaningless attributes and elements in modern real world updateinfo.xml files.

### CODE REVIEW

File: updateinfo/README.md [hunk 1]
```diff
@@ -44,8 +44,21 @@ and verify its correctness, and how to parse them.
...
+that its goal is to be representitive of what would be reasonable to expect
+in modern (2024) repositories.
 
-See [Permissive XML Schema for updateinfo.xml](updateinfo-permissive.xsd)
+See [Permissive XML Schema for updateinfo.xml](updateinfo-permissive.xsd) and
+[Transitional XML Schema for updateinfo.xml](updateinfo-transitional.xsd).
+
+**NOTE**: https://github.com/resf/distro-tools/issues/32 needs to be resolved
+before Rocky Linux updateinfo will be considered valid by the transitional
+schema.
```
Comment (Conan-Kudo) [inline updateinfo/README.md:58]: This information is unneeded here.

File: updateinfo/updateinfo-permissive.xsd [hunk 2]
```diff
@@ -100,6 +100,14 @@
 	    addition of a single element (reboot_suggested), and thus taking
 	    the second occurrence of the update element for a specific id as the
 	    canonical one would ensure correctness.
+
+            openSuSE 11 used the source package name as the update ID. In this
```
Comment (Conan-Kudo) [inline updateinfo/updateinfo-permissive.xsd:104]: `openSUSE`

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: updateinfo/README.md (hunk 1 above)
Comment (stewartsmith) [reply updateinfo/README.md]: fixed

File: updateinfo/updateinfo-permissive.xsd (hunk 2 above)
Comment (stewartsmith) [reply updateinfo/updateinfo-permissive.xsd]: fixed (and in another place it was there)

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**